File: src/main/java/itmo/is/project/dto/user/AccountDto.java
package itmo.is.project.dto.user;

import com.fasterxml.jackson.annotation.JsonProperty;
import io.swagger.v3.oas.annotations.media.Schema;

public record AccountDto (

        @JsonProperty(value = "user", required = true)
        UserDto user,

        @Schema(example = "1000000")
        @JsonProperty(value = "balance", required = true)
        Integer balance
) {
}

File: src/main/kotlin/itmo/isproject/shared/user/dto/AccountDto.kt
package itmo.isproject.shared.user.dto

import com.fasterxml.jackson.annotation.JsonProperty
import io.swagger.v3.oas.annotations.media.Schema

data class AccountDto(

    @JsonProperty(value = "user", required = true)
    val user: UserDto,

    @Schema(example = "1000000")
    @JsonProperty(value = "balance", required = true)
    val balance: Int?
)
